src/main/java/uk/gov/pay/ledger/transaction/dao/mapper/TransactionWithParentMapper.java

```diff
@@ -0,0 +1,92 @@
+                .withRefundStatus(rs.getString("parent_refund_status"))
+                .withRefundAmountRefunded(getLongWithNullCheck(rs, "parent_refund_amount_refunded"))
+                .withRefundAmountAvailable(getLongWithNullCheck(rs, "parent_refund_amount_available"))
+                .withFee(getLongWithNullCheck(rs, "parent_fee"))
+                .withTransactionType(rs.getString("parent_type"))
+                .build();
+
+        return new TransactionEntity.Builder()
+                .withId(rs.getLong("id"))
+                .withGatewayAccountId(rs.getString("gateway_account_id"))
+                .withExternalId(rs.getString("external_id"))
+                .withParentExternalId(rs.getString("parent_external_id"))
+                .withAmount(getLongWithNullCheck(rs, "amount"))
+                .withReference(rs.getString("reference"))
+                .withDescription(rs.getString("description"))
+                .withState(getState(rs, "state"))
+                .withEmail(rs.getString("email"))
+                .withCardholderName(rs.getString("cardholder_name"))
+                .withCreatedDate(getZonedDateTime(rs, "created_date").orElse(null))
+                .withTransactionDetails(rs.getString("transaction_details"))
+                .withEventCount(rs.getInt("event_count"))
+                .withCardBrand(rs.getString("card_brand"))
+                .withLastDigitsCardNumber(rs.getString("last_digits_card_number"))
+                .withFirstDigitsCardNumber(rs.getString("first_digits_card_number"))
+                .withNetAmount(getLongWithNullCheck(rs, "net_amount"))
+                .withTotalAmount(getLongWithNullCheck(rs, "total_amount"))
+                .withRefundStatus(rs.getString("refund_status"))
+                .withRefundAmountRefunded(getLongWithNullCheck(rs, "refund_amount_refunded"))
+                .withRefundAmountAvailable(getLongWithNullCheck(rs, "refund_amount_available"))
+                .withFee(getLongWithNullCheck(rs, "fee"))
+                .withTransactionType(rs.getString("type"))
+                .withParentTransactionEntity(parentTransactionEntity)
+                .build();
+    }
+
+    private TransactionState getState(ResultSet rs, String columnName) throws SQLException {
+        String state = rs.getString(columnName);
+        return Optional.ofNullable(state)
+                .map(TransactionState::valueOf)
+                .orElse(null);
+    }
+
+    private Long getLongWithNullCheck(ResultSet rs, String columnName) throws SQLException {
+        long value = rs.getLong(columnName);
+        return rs.wasNull() ? null : value;
+    }
+
+    private Optional<ZonedDateTime> getZonedDateTime(ResultSet rs, String columnLabel) throws SQLException {
+        Timestamp timestamp = rs.getTimestamp(columnLabel);
+
+        return Optional.ofNullable(timestamp)
+                .map(t -> ZonedDateTime.ofInstant(t.toInstant(), ZoneOffset.UTC));
+    }
```

Comment:
[improvement] Looks like we're duplicating methods from https://github.com/alphagov/pay-ledger/blob/6535ec6004839d3b986c6fbf8501ee052ce1c5e1/src/main/java/uk/gov/pay/ledger/transaction/dao/mapper/TransactionMapper.java (`getLongWithNullCheck` and `getZonedDateTime`). It would be nice to extract shared code.